offer-manager-domain/src/test/groovy/life/qbic/business/products/create/CreateProductSpec.groovy

```diff
@@ -0,0 +1,90 @@
+package life.qbic.business.products.create
+
+import life.qbic.business.exceptions.DatabaseQueryException
+import life.qbic.datamodel.dtos.business.ProductId
+import life.qbic.datamodel.dtos.business.services.AtomicProduct
+import life.qbic.datamodel.dtos.business.services.Product
+import life.qbic.datamodel.dtos.business.services.ProductUnit
+import spock.lang.Specification
+
+/**
+ * <h1>Tests for the CreateProduct use case</h1>
+ *
+ * <p>This specification contains tests for all steps of the create product use case</p>
+ *
+ * @since 1.0.0
+ */
+class CreateProductSpec extends Specification {
+    CreateProductOutput output
+    ProductId productId
+    Product product
+
+    def setup() {
+        output = Mock(CreateProductOutput)
+        productId = new ProductId("Test", "ABCD1234")
+        product = new AtomicProduct("test product", "this is a test product", 0.5, ProductUnit.PER_GIGABYTE, productId)
+    }
+
+    def "Create stores the provided product in the data source"() {
+        given: "a data source that stores a product"
+        CreateProductDataSource dataSource = Stub(CreateProductDataSource)
+        String dataStatus = ""
+        dataSource.store(product) >> { dataStatus = "stored" }
+        and: "an instance of the use case"
+        CreateProduct createProduct = new CreateProduct(dataSource, output)
+
+        when: "the create method is called"
+        createProduct.create(product)
+
+        then: "the output is informed and no failure notification is send"
+        1 * output.created(product)
+        0 * output.foundDuplicate(_)
+        0 * output.failNotification(_)
+        and: "the data was stored in the database"
+        dataStatus == "stored"
+    }
+
+    def "Create informs the output that an entry matching the provided product already exists"() {
+        given: "a data source that detects a duplicate entry"
+        CreateProductDataSource dataSource = Stub(CreateProductDataSource)
+        String dataStatus = ""
+        dataSource.store(product) >> {
+            dataStatus = "not stored"
+            println(dataStatus)
```

Comment:
```suggestion
```

Reply:
I would very much like to keep that line in the test to also have the output in the console.

Reply:
But this cannot be tested automatically. So I think it makes no sense to keep it.